pkg/controller/compliancesuite/suitererunner_cron_compat.go

```diff
@@ -0,0 +1,260 @@
+	if len(suiteName) >= 42 {
+		suiteName = suiteName[0:42]
+	}
+	return suiteName + "-rerunner"
+}
+
+func (r *ReconcileComplianceSuite) cronJobCompatCreate(
+	suite *compv1alpha1.ComplianceSuite,
+	key types.NamespacedName,
+	logger logr.Logger,
+) error {
+	var getObj client.Object
+
+	priorityClassName, err := r.getPriorityClassName(suite)
+	if err != nil {
+		logger.Error(err, "Cannot get priority class name, scan will not be run with set priority class")
+	}
+
+	createBeta := func() *batchv1beta1.CronJob {
+		getObj = &batchv1beta1.CronJob{}
+		return r.getBetaV1Rerunner(suite, priorityClassName)
+	}
+
+	createV1 := func() *batchv1.CronJob {
+		getObj = &batchv1.CronJob{}
+		return r.getV1Rerunner(suite, priorityClassName)
+	}
+
+	updateBeta := func() error {
+		getObjTyped, ok := getObj.(*batchv1beta1.CronJob)
+		if !ok {
+			return fmt.Errorf("failed to cast object to beta CronJob")
+		}
+		if getObjTyped.Spec.Schedule == suite.Spec.Schedule {
+			return nil
+		}
+		cronJobCopy := getObjTyped.DeepCopy()
+		cronJobCopy.Spec.Schedule = suite.Spec.Schedule
+		logger.Info("Updating beta rerunner", "CronJob.Name", cronJobCopy.GetName())
+		return r.Client.Update(context.TODO(), cronJobCopy)
+	}
+
+	updateV1 := func() error {
+		getObjTyped, ok := getObj.(*batchv1.CronJob)
+		if !ok {
+			return fmt.Errorf("failed to cast object to v1 CronJob")
+		}
+		if getObjTyped.Spec.Schedule == suite.Spec.Schedule {
+			return nil
+		}
+		cronJobCopy := getObjTyped.DeepCopy()
+		cronJobCopy.Spec.Schedule = suite.Spec.Schedule
+		logger.Info("Updating v1 rerunner", "CronJob.Name", cronJobCopy.GetName())
+		return r.Client.Update(context.TODO(), cronJobCopy)
+	}
+
+	createAction := func(o client.Object) error {
+		err := r.Client.Get(context.TODO(), key, getObj)
+		if err != nil && errors.IsNotFound(err) {
+			// No re-runner found, create it
+			logger.Info("Creating rerunner", "CronJob.Name", o.GetName())
+			return r.Client.Create(context.TODO(), o)
+		} else if err != nil {
+			return err
+		}
+
+		switch o.(type) {
+		case *batchv1beta1.CronJob:
+			return updateBeta()
+		case *batchv1.CronJob:
+			return updateV1()
+		}
+
+		return nil
+	}
+
+	return doCompat(createAction, createBeta, createV1)
+}
+
+func cronJobCompatGet(r *ReconcileComplianceSuite, key types.NamespacedName) (client.Object, error) {
+	var retObj client.Object
+
+	getEmptyBeta := func() *batchv1beta1.CronJob {
+		return &batchv1beta1.CronJob{}
+	}
+
+	getEmptyV1 := func() *batchv1.CronJob {
+		return &batchv1.CronJob{}
+	}
+
+	getAction := func(o client.Object) error {
+		err := r.Client.Get(context.TODO(), key, o)
+		if err != nil && errors.IsNotFound(err) {
+			// No re-runner found, we're good
+			return nil
+		} else if err != nil {
+			return err
+		}
+
+		retObj = o
+		return nil
+	}
+
+	err := doCompat(getAction, getEmptyBeta, getEmptyV1)
+	return retObj, err
+}
+
+func cronJobCompatDelete(r *ReconcileComplianceSuite, cron client.Object) error {
+	if cron == nil {
```

Comment:
Question: Should we return `errors.IsNotFound` from `cronJobCompatGet` and let the caller handle that error rather than handling nil here? Or is that just bike shedding?